pkg/storage/stores/indexshipper/compactor/deletion/delete_requests_manager.go

```diff
@@ -23,7 +23,7 @@ const (
 )
 
 type userDeleteRequests struct {
-	requests []DeleteRequest
+	requests []*DeleteRequest
```

Comment:
Why is a pointer preferable?

Reply:
because we are losing updates due to working on different copies; for example, we are always printing 0 lines deleted even when we have incremented it here https://github.com/grafana/loki/blob/c3e12d76a0144ba9d060bb972e070b000789fb81/pkg/storage/stores/indexshipper/compactor/deletion/delete_request.go#L72